Library/ViewModels/CuratedProjectsViewModelTests.swift

```diff
@@ -0,0 +1,53 @@
+@testable import KsApi
+@testable import Library
+import Prelude
+import ReactiveExtensions
+import ReactiveExtensions_TestHelpers
+import ReactiveSwift
+import XCTest
+
+final class CuratedProjectsViewModelTests: TestCase {
+  private let dismissViewController = TestObserver<Void, Never>()
+  private let loadProjects = TestObserver<[Project], Never>()
+  private let viewModel: CuratedProjectsViewModelType = CuratedProjectsViewModel()
+
+  override func setUp() {
+    super.setUp()
+    self.viewModel.outputs.dismissViewController.observe(self.dismissViewController.observer)
+    self.viewModel.outputs.loadProjects.observe(self.loadProjects.observer)
+  }
+
+  func testDismissViewController_OnButtonTap() {
+    self.dismissViewController.assertDidNotEmitValue()
+
+    self.viewModel.inputs.doneButtonTapped()
+
+    self.dismissViewController.assertValueCount(1)
+  }
+
+  func testLoadProjects() {
+    let artProjects = (1...5).map {
+      .template
+        |> Project.lens.id .~ $0
+        |> Project.lens.category .~ Project.Category.art
+    }
+    let gamesProjects = (1...5).map {
+      .template
+        |> Project.lens.id .~ $0
+        |> Project.lens.category .~ Project.Category.games
+    }
+    let projects = (artProjects + gamesProjects)
+
+    let envelope = .template
+      |> DiscoveryEnvelope.lens.projects .~ (artProjects + gamesProjects)
```

Comment:
This doesn't exactly reflect what we expect to happen. What we expect to happen is for two independent network requests to be made with the responses being `projects .~ artProjects` and `projects .~ gamesProjects`. Since we don't have a way currently to mock out different responses on the same endpoint, I would say it's enough to check that the number of projects returned to `loadProjects` is correct, because that will tell us how many times the request was made. In that case, all you need is something like this:

```    
  let projects = (1...5).map {
      .template
        |> Project.lens.id .~ $0
    }

    let envelope = .template
      |> DiscoveryEnvelope.lens.projects .~ projects

    let apiService = MockService(fetchDiscoveryResponse: envelope)

    withEnvironment(apiService: apiService) {
      self.viewModel.inputs.configure(with: [Category.art, Category.tabletopGames])

      self.viewModel.inputs.viewDidLoad()

      self.loadProjects.assertValues([projects + projects]) // 2x the projects, shows request was made 2x
  }
```